lib/msal-core/src/utils/RequestUtils.ts

```diff
@@ -0,0 +1,145 @@
+import { Constants, TemporaryCacheKeys, PromptState, BlacklistedEQParams } from "../utils/Constants";
+import { AuthCache } from "../cache/AuthCache";
+import { ClientAuthError } from "../error/ClientAuthError";
+import { ClientConfigurationError } from "../error/ClientConfigurationError";
+import { ScopeSet } from "../ScopeSet";
+import { StringDict } from "../MsalTypes";
+import { UrlUtils } from "../utils/UrlUtils";
+import { WindowUtils } from "../utils/WindowUtils";
+import { StringUtils } from "../utils/StringUtils";
+import { CryptoUtils } from "../utils/CryptoUtils";
+
+/**
+ * @hidden
+ */
+export class RequestUtils {
+
+    /**
+     * @ignore
+     *
+     * @param request
+     * @param isLoginCall
+     * @param requestType
+     * @param redirectCallbacksSet
+     * @param cacheStorage
+     * @param clientId
+     *
+     * validates all request parameters and generates a consumable request object
+     */
+    static validateRequest(request: AuthenticationParameters, isLoginCall: boolean, requestType: string, redirectCallbacksSet: boolean, cacheStorage: AuthCache, clientId: string): AuthenticationParameters {
+
+        let validatedRequest: AuthenticationParameters;
+
+        // Throw error if request is empty for acquire * calls
+        if(!isLoginCall && !request) {
+            throw ClientConfigurationError.createEmptyRequestError();
+        }
+
+        // Throw error if callbacks are not set before redirect
+        if(requestType == Constants.interactionTypeRedirect) {
+            if (!redirectCallbacksSet) {
+                cacheStorage.removeItem(TemporaryCacheKeys.INTERACTION_STATUS);
+                throw ClientConfigurationError.createRedirectCallbacksNotSetError();
+            }
+        }
+
+        // if extraScopesToConsent is passed in loginCall, append them to the login request; Validate and filter scopes (the validate function will throw if validation fails)
+        const scopes: Array<string> = isLoginCall ? ScopeSet.appendScopes(request.scopes, request.extraScopesToConsent) : request && request.scopes;
+        ScopeSet.validateInputScope(scopes, !isLoginCall, clientId);
+
+        // validate prompt parameter
+        this.validatePromptParameter(request && request.prompt);
+
+        // validate extraQueryParameters and claimsRequest
+        this.validateEQParameters(request && request.extraQueryParameters, request && request.claimsRequest);
+
+        const state = this.validateAndGenerateState(request && request.state);
+        const correlationId = this.validateAndGenerateCorrelationId(request && request.correlationId);
+
+        return validatedRequest = {
+            ...request,
+            scopes,
+            state,
+            correlationId
+        };
+    }
+
+    /**
+     * @ignore
+     *
+     * blocks any login/acquireToken calls to reload from within a hidden iframe (generated for silent calls)
+     */
+    static blockReloadInHiddenIframes() {
```

Comment:
Probably makes more sense in WindowUtils.

Reply:
Changed this.